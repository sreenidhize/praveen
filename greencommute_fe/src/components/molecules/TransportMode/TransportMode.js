import React from "react";
import { makeStyles, Typography } from "@material-ui/core";
import PropTypes from "prop-types";
import DirectionsBusIcon from "@material-ui/icons/DirectionsBus";
import DirectionsWalk from "@material-ui/icons/DirectionsWalk";
import TrainOutlinedIcon from "@material-ui/icons/TrainOutlined";

const useStyles = makeStyles((theme) => ({
  root: {
    height: "20px",
    width: "20px",
    color: theme.typography.body2.color,
    marginTop: 3,
  },
  type: {
    display: "flex",
    alignItems: "flex-start",
    paddingTop: theme.spacing(0.5),
    paddingBottom: theme.spacing(0.5),
    color: theme.typography.body2.color,
  },
  desc: {
    paddingTop: theme.spacing(1),
    marginLeft: theme.spacing(0.5),
    color: theme.typography.body2.color,
  },
  status: {
    marginLeft: theme.spacing(1),
    color: theme.typography.overline.color,
  },
  onTimeStatus: {
    color: "#6bea9a",
    marginLeft: theme.spacing(1),
  },
  transitDesc: {
    marginBottom: theme.spacing(1),
    color: theme.typography.body2.color,
  },
  fare: {
    color: theme.typography.body2.color,
  },
}));

const TransportMode = (props) => {
  const classes = useStyles();
  const { type, description, fare, time, status, isLate } = props;
  return (
    <div className={classes.type} data-testid="transit-type">
      <div>
        {type === "metro" ? (
          <TrainOutlinedIcon className={classes.root} />
        ) : type === "bus" ? (
          <DirectionsBusIcon className={classes.root} />
        ) : (
          <DirectionsWalk className={classes.root} />
        )}
      </div>
      <div item className={classes.desc}>
        <Typography variant={"subtitle2"} className={classes.transitDesc}>
          {description}
          {!!time && type === "walk" ? (
            <span> {time}</span>
          ) : isLate ? (
            <span className={classes.status}>{status}</span>
          ) : (
            <span className={classes.onTimeStatus}>On time</span>
          )}
        </Typography>
        {!!fare && !!time && (
          <Typography variant={"subtitle2"} className={classes.fare}>
            {fare} &bull; {time}
          </Typography>
        )}
      </div>
    </div>
  );
};

TransportMode.propTypes = {
  type: PropTypes.string,
  fare: PropTypes.string,
  time: PropTypes.string,
  description: PropTypes.string.isRequired,
  status: PropTypes.string,
  isLate: PropTypes.bool,
};

export default TransportMode;
