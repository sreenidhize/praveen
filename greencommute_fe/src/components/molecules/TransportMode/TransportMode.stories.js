import React from 'react';
import TransportMode from "./TransportMode";

export default {
    title: 'Molecules/DetailedIconText',
    component: TransportMode
}

const Template = (args) => <TransportMode {...args} />;

export const Text1 = Template.bind({});
Text1.args = {
    type: 'bus',
    fare: '15$',
    time: '20 mins',
    description: "Catch a bus 38X at 3.42 PM to Secunderabad bus stand.",
    status: '2 mins',
    isLate: true
};

export const Text2 = Template.bind({});
Text2.args = {
    type: 'walk',
    time: '5 mins',
    description: "Head north on St. Johns road."
};

export const Text3 = Template.bind({});
Text3.args = {
    type: 'metro',
    fare: '56$',
    time: '38 mins',
    description: "Catch a blue line metro towards Raidurg.",
    isLate: false
};