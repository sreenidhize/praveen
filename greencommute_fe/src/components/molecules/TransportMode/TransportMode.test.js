import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import TransportMode from "./TransportMode";

describe("TransportMode", () => {
    it("Matches the snapshot", () => {
        const { container } = render(
            <TransportMode description={"Walk for 5 mins"}/>
        );
        expect(container).toMatchSnapshot();
    });

    test("component renders correctly", () => {
        const { getByTestId } = render(<TransportMode description={"Walk for 5 mins"}/>);
        expect(getByTestId("transit-type")).toBeInTheDocument();
    });
});