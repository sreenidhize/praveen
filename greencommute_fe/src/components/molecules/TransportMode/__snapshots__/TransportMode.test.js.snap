// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransportMode Matches the snapshot 1`] = `
<div>
  <div
    class="makeStyles-type-2"
    data-testid="transit-type"
  >
    <div>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-root-1"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M13.5 5.5c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zM9.8 8.9L7 23h2.1l1.8-8 2.1 2v6h2v-7.5l-2.1-2 .6-3C14.8 12 16.8 13 19 13v-2c-1.9 0-3.5-1-4.3-2.4l-1-1.6c-.4-.6-1-1-1.7-1-.3 0-.5.1-.8.1L6 8.3V13h2V9.6l1.8-.7"
        />
      </svg>
    </div>
    <div
      class="makeStyles-desc-3"
    >
      <h6
        class="MuiTypography-root makeStyles-transitDesc-6 MuiTypography-subtitle2"
      >
        Walk for 5 mins
        <span
          class="makeStyles-onTimeStatus-5"
        >
          On time
        </span>
      </h6>
    </div>
  </div>
</div>
`;
